ledger/participant-integration-api/src/main/scala/platform/index/ReadOnlySqlLedgerWithMutableCache.scala

```diff
@@ -52,13 +58,18 @@ private[index] object ReadOnlySqlLedgerWithMutableCache {
         (ledgerEndOffset, ledgerEndSequentialId) <- Resource.fromFuture(
           ledgerDao.lookupLedgerEndOffsetAndSequentialId()
         )
+        _ = ledgerEndCache.set((ledgerEndOffset, ledgerEndSequentialId))
         prefetchingDispatcher <- dispatcherOffsetSeqIdOwner(
           ledgerEndOffset,
           ledgerEndSequentialId,
         ).acquire()
         generalDispatcher <- dispatcherOwner(ledgerEndOffset).acquire()
         dispatcherLagMeter <- Resource.successful(
-          new DispatcherLagMeter(generalDispatcher.signalNewHead)(
+          new DispatcherLagMeter((offset, eventSeqId) => {
+            ledgerEndCache.set((offset, eventSeqId))
+            // the order here is very important: first we need to make stuff available for point-wise lookups and SQL queries, and only then we can make it available on the streams. (consider example: completion arrived on a stream, but the transaction cannot be looked up)
```

Comment:
Consider breaking comments to fit within a max line length to aid reviewers using Github as the UI.